apps/test/applabTest.js

```diff
@@ -48,3 +49,71 @@ describe('designMode.addScreenIfNecessary', function () {
     assert.equal(converted, html);
   });
 });
+
+describe('getIdDropdown filtering modes', function () {
+  var documentRoot;
+
+  beforeEach(function () {
+    // Create a sample DOM to test against
+    var sampleDom =
+        '<div>' +
+          '<div id="divApplab" class="appModern">' +
+            '<div class="screen" id="screen1">' +
+              '<div class="chart" id="chart9"></div>' +
+              '<img src="" class="chart-friend" id="image1">' +
+            '</div>' +
+          '</div>' +
+        '</div>';
+    documentRoot = $(sampleDom);
+  });
+
+  it('produces all IDs when no filter is given', function () {
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot), [
+      { "display": '"chart9"', "text": '"chart9"' },
+      { "display": '"image1"', "text": '"image1"' },
+      { "display": '"screen1"', "text": '"screen1"' }
+    ]);
+  });
+
+  it('can filter on tag type', function () {
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, 'div'), [
+      { "display": '"chart9"', "text": '"chart9"' },
+      { "display": '"screen1"', "text": '"screen1"' }
+    ]);
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, 'img'), [
+      { "display": '"image1"', "text": '"image1"' }
+    ]);
+  });
+
+  it('can filter on class', function () {
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, '.chart'), [
+      { "display": '"chart9"', "text": '"chart9"' }
+    ]);
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, '.screen'), [
+      { "display": '"screen1"', "text": '"screen1"' }
+    ]);
+  });
+
+  it('can filter on ID', function () {
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, '#screen1'), [
+      { "display": '"screen1"', "text": '"screen1"' }
+    ]);
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, '#chart9'), [
+      { "display": '"chart9"', "text": '"chart9"' }
+    ]);
+    assert.deepEqual(AppLab.getIdDropdownFromDom_(documentRoot, '#image1'), [
+      { "display": '"image1"', "text": '"image1"' }
+    ]);
+  });
+
+  it('does not accidentally pick up superset classes', function () {
```

Comment:
FYI, I'm leaving these tests in even though we're switching to the jQuery implementation.  It never hurts to prove that your libraries do what you expect them to :wink:.